lambdas/user-reverification/src/test/java/uk/gov/di/ipv/core/userreverification/pact/UserReverificationHandlerTest.java

```diff
@@ -0,0 +1,129 @@
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class UserReverificationHandlerTest {
+    private static final String IPV_SESSION_ID = "mockIpvSessionId";
+    private static final String CLIENT_OAUTH_SESSION_ID = "mockClientOAuthSessionId";
+    private static final String REVERIFICATION_SCOPE = "reverification";
+
+    private LambdaHttpServer httpServer;
+
+    @Mock private Sleeper mockSleeper;
+    @Mock private DataStore<IpvSessionItem> mockIpvSessionDataStore;
+    @Mock private ConfigService mockConfigService;
+    @Mock private DataStore<ClientOAuthSessionItem> mockOAuthSessionStore;
+    @Mock private SessionCredentialsService mockSessionCredentialsService;
+
+    @PactBrokerConsumerVersionSelectors
+    public static SelectorBuilder consumerVersionSelectors() {
+        return new SelectorBuilder().mainBranch();
+    }
+
+    @BeforeAll
+    static void setupServer() {
+        System.setProperty("pact.content_type.override.application/jwt", "text");
+    }
+
+    @BeforeEach
+    void pactSetup(PactVerificationContext context) throws IOException {
+        var ipvSessionService = new IpvSessionService(mockIpvSessionDataStore, mockSleeper);
+        var clientOAuthSessionDetailsService =
+                new ClientOAuthSessionDetailsService(mockOAuthSessionStore);
+
+        var handler =
+                new UserReverificationHandler(
+                        ipvSessionService,
+                        mockConfigService,
+                        clientOAuthSessionDetailsService,
+                        mockSessionCredentialsService);
+
+        httpServer = new LambdaHttpServer(handler, "/reverification");
+        httpServer.startServer();
+
+        context.setTarget(new HttpTestTarget("localhost", httpServer.getPort()));
+    }
+
+    @AfterEach
+    public void tearDown() {
+        httpServer.stopServer();
+    }
+
+    @State("accessToken is a valid access token")
+    public void setAccessToken() {
+        var ipvSession = new IpvSessionItem();
+        ipvSession.setIpvSessionId(IPV_SESSION_ID);
+        ipvSession.setClientOAuthSessionId("mockClientOAuthSessionId");
+        ipvSession.setAccessTokenMetadata(new AccessTokenMetadata());
+
+        var oAuthSession = new ClientOAuthSessionItem();
+        oAuthSession.setUserId("mockUserId");
+        oAuthSession.setClientId("mockClientId");
+        oAuthSession.setGovukSigninJourneyId(CLIENT_OAUTH_SESSION_ID);
+        oAuthSession.setScope(REVERIFICATION_SCOPE);
+
+        when(mockOAuthSessionStore.getItem(CLIENT_OAUTH_SESSION_ID)).thenReturn(oAuthSession);
+        when(mockIpvSessionDataStore.getItemByIndex(
+                        "accessToken", DigestUtils.sha256Hex("accessToken")))
+                .thenReturn(ipvSession);
+    }
+
+    @State("accessToken is a invalid access token")
+    public void dontSetAccessToken() {
+        /*
+            This method is empty as it doesn't set an access token in order to invoke a 403
+        */
```

Comment:
I might rephrase this slightly (it took me a second to understand...)

Perhaps:
> This method is empty - access tokens are invalid by default

The point is that we're meant to set up `accessToken` as an invalid access token, whether it produces a 403 is what's being tested by the pact.